src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2224,6 +2239,49 @@ export class StudyViewPageStore
         }
     }
 
+    parseRawFilterJson(filterJson: string): any {
+        let rawJson;
+        try {
+            rawJson = JSON.parse(decodeURIComponent(filterJson));
+        } catch (e) {
+            console.error('FilterJson invalid Json: error: ', e);
+        }
+        return rawJson;
+    }
+
+    async fetchSamplesWithStudyIds(studyIds: string[]) {
+        let studyViewFilter: StudyViewFilter = {} as any;
+        studyViewFilter.studyIds = studyIds;
+        return await internalClient.fetchFilteredSamplesUsingPOST({
+            studyViewFilter: studyViewFilter,
+        });
+    }
+
+    convertPatientIdentifiersToSampleIdentifiers(
+        patientIdentifierFilter: PatientIdentifierFilter,
+        samples: Sample[]
+    ): SampleIdentifier[] {
+        const sampleIdentifiers: SampleIdentifier[] = [];
+        patientIdentifierFilter.patientIdentifiers.forEach(
```

Comment:
i think this can be optimized and made clearer by reversing the loop.  
```
const sampleIdentifiers = samples.filter(sample=>i belong to a patient in patientIdetnifiers).map(...)
```
further optimization would be to make a dictionary of patients with compoundKey.